rsconf/component/__init__.py

```diff
@@ -332,6 +329,38 @@ def _bash_append_and_dst(
         self._bash_append(host_path, md5=md5)
         return dst
 
+    def _gen_host_and_identity_ssh_keys(
+        self, j2_ctx, filename, visibility, password=None
+    ):
+        from rsconf import db
+
+        res = PKDict()
+        b = db.secret_path(j2_ctx, filename, visibility=visibility, directory=True)
+        res.host_key_f = b.join("host_key")
+        res.host_key_pub_f = res.host_key_f + ".pub"
```

Comment:
I think `b.new(ext="pub")` is how you are supposed to do this in py.path.

Reply:
0b2eeca